manual/01-intro/Immutability.md

```diff
@@ -1,4 +1,168 @@
+
+In functional programming, such mutable state is considered extremely
+dangerous, making programs hard to reason about, difficult to debug,
+and almost impossible to grow correctly into large, consistent systems.
+
+Ramda helps avoid this by refusing to ever mutate input data.  We have
+equivalents of the accessors above:
+
+```js
+var getName = R.prop('name');
+getName(person); //=> "Fred"
+// or 
+R.prop('name', person); //=> "Fred"
+```
+
+But the closest equivalents we have to mutators are functions like 
+`assoc`.  And they're not all that close.  `assoc` creates a copy of
+your object with a new value associated with a field name:
+
+```js
+var barney = {name: "Barney", age: 27}
+var fred = {name: "Fred", age: 29}
+barney.bestFriend = fred;
+fred.bestFriend = barney;
+
+var fred2 = R.assoc('age', 30, fred);
+
+fred2.age; //=> 30
+fred.age; //> 29
+fred2.bestFriend === barney; //=> true
+```
+
+Note that `assoc` retuns a new object, but note also the last line; the
+returned structure shares as much of its internals with the original as
+possible.  This is not a deep clone of the original object.
+
+This is true across all of Ramda.  Nothing you pass to Ramda will be 
+altered.  If a function acts like a mutator, it instead creates a copy
+of the structure with the changes applied:
+
+For example, the native `sort` and `reverse` methods on`Array.prototype`, 
+mutate the underlying object.  Ramda's equivalents just return new lists:
+
+```js
+var a1 = [1, 2, 3];
+var b1 = a1.reverse(); //=> [3, 2, 1]
+a1; //=> [3, 2, 1]
+
+var a2 = [1, 2, 3];
+var b2 = R.reverse(a2); //=> [3, 2, 1]
+a2; //=> [1, 2, 3]
+
+// and similarly for `sort`
+```
+
+We won't discuss in depth all the reasons that immutability is desirable.
+That's easily available [on the web][go].  But there are many important
+reasons having to do with how well anyone can understand our code, how
+well it stands up to parallelization, how many fewer ways in which it can
+actually fail.
+
+We also will not talk about how you might make more of our own data
+immutable (although in [Chapter 10][im], we do discuss how Ramda will
+integrate with some immutable data libraries but not others.)  But we do
+want to stress that _none_ of Ramda's functions mutate user data.
+
+For those used to functional programming, we should point out that this 
+is not done through deeply recursive algorithms.  We have plenty of 
+internal mutation.  But at our function boundaries, everything can be 
+seen as immutable.
+
+The basic point is simple: Don't expect Ramda to mutate your data for
+you:
+
+```js
+var numbers = [8, 6, 7, 5, 3, 0];
+var jenny = R.append(9, numbers);
+numbers; //=> [8, 6, 7, 5, 3, 0]
+jenny; //=> [8, 6, 7, 5, 3, 0, 9]
+```
+
+Ramda does not take any special pains to prevent you from mutating
+your own data when using its functions.  For intance you could build 
+up an ouput object by continually mutating the accumulator passed to
```

Comment:
Yup.  Will fix when I'm next in there.  I haven't tried to see if we can have Travis build these, especially as you said you wanted to updated the docs build.  But it would be nice to get there at some point.